go/pkg/apis/enricher/framework/go/go_detector.go

```diff
@@ -28,6 +30,44 @@ func hasFramework(modules []*modfile.Require, tag string) bool {
 	return false
 }
 
+func DoGoPortsDetection(component *model.Component, ctx *context.Context) {
+	files, err := utils.GetCachedFilePathsFromRoot(component.Path, ctx)
+	if err != nil {
+		return
+	}
+
+	matchRegexRules := model.PortMatchRules{
+		MatchIndexRegexes: []model.PortMatchRule{
+			{
+				Regex:     regexp.MustCompile(`.ListenAndServe\([^,)]*`),
+				ToReplace: ".ListenAndServe(",
+			},
+			{
+				Regex:     regexp.MustCompile(`.Start\([^,)]*`),
```

Comment:
Just curious, what command would this match on? Does go have a Start function that uses ports

Reply:
As it is mentioned here: https://github.com/redhat-developer/alizer/blob/main/go/docs/port_detection.md#echo we are looking for two methods `ListenAndServe` & `Start`. Not related but as we have been focusing to port detection, I've created an issue for the update of this document https://github.com/redhat-developer/alizer/issues/185